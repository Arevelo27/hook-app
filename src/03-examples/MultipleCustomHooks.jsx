import { useFetch } from "../hooks/"

export const MultipleCustomHooks = () => {

    const { data, isLoading, hasError } = useFetch('https://pokeapi.co/api/v2/pokemon/3');

    return (
        <div>
            <h1>Información de Pokemón</h1>
            <hr />

            {isLoading && <h2>Cargando...</h2>}
            {hasError && <h2>{hasError}</h2>}

            <h2>{data?.name}</h2>
            {/* <pre>{JSON.stringify(data, null, 2)}</pre> */}
        </div>
    )
}
