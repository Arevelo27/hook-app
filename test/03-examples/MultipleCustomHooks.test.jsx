import { render, screen } from '@testing-library/react';
import { MultipleCustomHooks } from '../../src/03-examples/MultipleCustomHooks';

describe('Pruebas en el <MultipleCustomHooks>', () => {

    test('debe de hacer match con el snapshot', () => {
        const { container } = render(<MultipleCustomHooks />);
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar el componente <MultipleCustomHooks>', () => {
        render(<MultipleCustomHooks />);
        expect(screen.getByText('Información de Pokemón')).toBeTruthy();
        expect(screen.getByText('Cargando')).toBeTruthy();

        const nextButton = screen.getByRole('button', { name: 'Siguiente' });
        expect(nextButton).toBeTruthy();
        expect(nextButton.disabled).toBeTruthy();

        screen.debug();
    })
  
})