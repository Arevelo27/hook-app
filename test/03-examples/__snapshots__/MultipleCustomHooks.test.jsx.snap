// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <MultipleCustomHooks> debe de hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1>
      Información de Pokemón
    </h1>
    <hr />
    <section
      class="alert alert-info text-center p-0 m-0 d-flex justify-content-center align-items-center"
      style="height: 200px;"
    >
      <h1>
        Cargando
      </h1>
    </section>
    <h2 />
    <button
      class="btn btn-primary mt-2"
    >
      Anterior
    </button>
    <button
      class="btn btn-primary mt-2"
      disabled=""
    >
      Siguiente
    </button>
  </div>
</div>
`;
